cloudnet/src/main/java/de/dytanic/cloudnet/network/listener/cluster/PacketServerSyncTemplateStorageListener.java

```diff
@@ -0,0 +1,88 @@
+package de.dytanic.cloudnet.network.listener.cluster;
+
+import com.google.common.base.Preconditions;
+import de.dytanic.cloudnet.CloudNet;
+import de.dytanic.cloudnet.driver.api.DriverAPIRequestType;
+import de.dytanic.cloudnet.driver.network.INetworkChannel;
+import de.dytanic.cloudnet.driver.network.protocol.IPacket;
+import de.dytanic.cloudnet.driver.network.protocol.IPacketListener;
+import de.dytanic.cloudnet.driver.serialization.ProtocolBuffer;
+import de.dytanic.cloudnet.driver.service.ServiceTemplate;
+import de.dytanic.cloudnet.driver.template.TemplateStorage;
+import de.dytanic.cloudnet.template.ClusterSynchronizedTemplateStorage;
+
+public class PacketServerSyncTemplateStorageListener implements IPacketListener {
+
+    private final boolean redirectToCluster;
+
+    public PacketServerSyncTemplateStorageListener(boolean redirectToCluster) {
+        this.redirectToCluster = redirectToCluster;
+    }
+
+    @Override
+    public void handle(INetworkChannel channel, IPacket packet) throws Exception {
+        ProtocolBuffer buffer = packet.getBuffer();
+
+        DriverAPIRequestType requestType = buffer.readEnumConstant(DriverAPIRequestType.class);
+        ServiceTemplate template = buffer.readObject(ServiceTemplate.class);
+
+        TemplateStorage templateStorage = CloudNet.getInstance().getTemplateStorage(template.getStorage());
+        Preconditions.checkNotNull(templateStorage, "invalid storage %s", template.getStorage());
+
+        if (!(templateStorage instanceof ClusterSynchronizedTemplateStorage)) {
+            throw new IllegalArgumentException("TemplateStorage " + template.getStorage() + " is not cluster synchronized");
+        }
+        ClusterSynchronizedTemplateStorage storage = (ClusterSynchronizedTemplateStorage) templateStorage;
+
+        switch (requestType) {
```

Comment:
Codacy found an issue: [Switch statements should have a default label](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=6033851)